### PR TITLE
Add localization support for Plus Installer

### PR DESCRIPTION
**Pull request by yfdyh000**
I basically tested it with Inno Setup 6.2.0.

It brings localized context menu in explorer.
A small flaw is an overlay upgrade will bring repetitive files for shortcut name changes (uninstalling can clear them).
The `Sandboxie-Plus Website` string has changed, use the official original and translations, the "Start when Windows starts" too.

### CODE REVIEW

File: Installer/Sandboxie-Plus.iss [hunk 1]
```diff
@@ -46,18 +46,55 @@ Source: ".\Sandboxie-Plus.ini"; DestDir: "{app}"; Flags: ignoreversion onlyifdoe
 
 [Icons]
 Name: "{group}\Sandboxie-Plus"; Filename: "{app}\SandMan.exe"; MinVersion: 0.0,5.0;
-Name: "{group}\Sandboxie-Plus Website"; Filename: "http://sandboxie-plus.com/"; MinVersion: 0.0,5.0;
-Name: "{group}\Uninstall Sandboxie-Plus"; Filename: "{uninstallexe}"; MinVersion: 0.0,5.0;
+Name: "{group}\{cm:ProgramOnTheWeb,{#MyAppName}}"; Filename: "http://sandboxie-plus.com/"; MinVersion: 0.0,5.0;
+Name: "{group}\{cm:UninstallProgram,{#MyAppName}}"; Filename: "{uninstallexe}"; MinVersion: 0.0,5.0;
```
Comment (isaak654) [inline Installer/Sandboxie-Plus.iss:50]: I don't know if this uninstall string is intended to be hidden from here:
immagine

Comment (yfdyh000) [reply Installer/Sandboxie-Plus.iss:50]: I can't reproduce the lack of the "uninstall" shortcut.

Comment (isaak654) [reply Installer/Sandboxie-Plus.iss:50]: The "uninstall" shorcut appeared to me only for the very first moments after the installation, but it tends to disappear soon after opening/closing "Start" menu multiple times.

Comment (isaak654) [reply Installer/Sandboxie-Plus.iss:50]: According to this, it seems a W10 thing: https://devblogs.microsoft.com/oldnewthing/20190319-00/?p=102342

**Comment by isaak654** [comment]
@mpheath
Unfortunately there isn't another review about these Inno Setup changes, do you think this could be improved furtherly or it's fine?

**Comment by mpheath** [comment]
@isaak654  
This PR looks very interesting. It appears as I would expect as being correct for multilingual. It looks good to me.
After 2 am at moment...tired.  I will try to give it a test tomorrow.

**Comment by mpheath** [comment]
Done some testing and have some notes:

 * Missing `Folder` key custom message text for context menu registration.
   The Folder key was added *2021-07-06* after the PR was pushed on *2021-06-19*.
   The custom message is same as the `*` key.
 * Custom messages warnings of *not defined* is large. Compiles OK, just lots of warnings.
   If a custom message is not set for the selected language, then the default language custom message is used, which is OK.
 * Sandman.exe is overwriting the context menu entries which changes the custom message text to english.
   As a possible fix, Sandman.exe can check if the `sandbox` key (or the value for the custom message) exists and not write if it does exist.
   Seems like a InnoSetup vs. Sandman battle over the registry :)

Test with Innosetup 6.2.0. Installed as Italian and French. I am basically happy with it. Some fixes may need to happen after commit.

**Comment by mpheath** [comment]
I have a alternative [innosetup script](https://github.com/mpheath/Sandboxie/blob/innosetup_alt/Installer/Sandboxie-Plus.iss) at my forked repo. It has 1 CustomMessage section. I have moved some code into other sections to cleanup the code section. I added the multilingual custom messages and language files of this PR. This alternative can be PRed later if acceptable. It has tested OK so far, though it is still new. It has implemented the ideas requested from #1036 for removing Sandboxie.ini...

If this PR can be somewhat similar, then an easy update might be possible for an alternative. Or perhaps some ideas could be taken from the alternative. Just swimming in the options.

**Comment by mpheath** [comment]
If can be done in this PR, or after the commit, is make a include file *Languages.iss* and move the *Languages* section and the *CustomMessages* section into it. The `#include "Languages.iss"` line is added to *Sandboxie-Plus.iss*. This allows the language translators to edit *Languages.iss* instead of the main *Sandboxie-Plus.iss*. Less confusing for the translators and leaves the *Sandboxie-Plus.iss* to updating less often. View the 2 files [here](https://github.com/mpheath/Sandboxie/tree/innosetupi18n/Installer) which have been tested and show what I just mentioned.

**Comment by isaak654** [comment]
I've just merged your changes, but I think this could be furtherly improved by commenting out the lines of all languages that don't have a defined Plus translation and adding new isl files only for current Plus translations. In this way, all translators will find out the contrast with the few entries that are still in english, and they will know that they can be translated.

Another thing I would like to have an opinion is the introduction of a language selector when you run Plus installer, like in Classic installer:
immagine

**Comment by mpheath** [comment]
@isaak654 
InnoSetup does the selector automatically once multiple languages are available! ;)

image

Try a test of  [Sandboxie-Plus-x64-v0.8.8.exe](http://users.tpg.com.au/mpheath/support/Sandboxie-Plus-x64-v0.8.8.exe) in a VM. It is not signed though works in Windows 10, not in Windows 7 as I guess it needs to be signed to install the Sbie driver. I'll delete this once you got the download, so a stray test installer does not do the rounds.

**Comment by isaak654** [comment]
> 
> 
> InnoSetup does the selector automatically once multiple languages are available! ;)
> 

That's a relief! I thought that ShowLanguageDialog parameter was required.

> Try a test of [Sandboxie-Plus-x64-v0.8.8.exe](http://users.tpg.com.au/mpheath/support/Sandboxie-Plus-x64-v0.8.8.exe) in a VM. It is not signed though works in Windows 10, not in Windows 7 as I guess it needs to be signed to install the Sbie driver. I'll delete this once you got the download, so a stray test installer does not do the rounds.

I got it, thank you!
I'm going to comment out the unsupported languages and see if I can find all other isl files.

**Comment by isaak654** [comment]
It looks fine now, I just struggled a bit with git.
If you don't notice anything else, I'll mark this as ready.

I'll add a note in the changelog when this pull request will be merged.

**Review by isaak654 — APPROVED** [review]
Tested and ready to be merged

**Comment by isaak654** [comment]
@mpheath 
David informed me that the today 0.9.0 release will not use these changes as it was already signed, but it will be used with future builds. I'm looking forward to see the new pull request with the uninstall improvements.

**Pull request merged**